filesmanager/filesmanager.py

```diff
@@ -48,20 +132,236 @@ def student_view(self, context=None):
-    @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    def studio_view(self, context=None):
         """
-        An example handler, which increments the data.
+        The edit view of the FilesManagerXBlock in Studio.
         """
-        if suffix:
-            pass  # TO-DO: Use the suffix when storing data.
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+        if context:
+            pass  # TO-DO: do something based on the context.
+        html = self.resource_string("static/html/filesmanager.html")
+        frag = Fragment(html.format(self=self))
+        frag.add_css(self.resource_string("static/css/filesmanager.css"))
+
+        # Add i18n js
+        statici18n_js_url = self._get_statici18n_js_url()
+        if statici18n_js_url:
+            frag.add_javascript_url(self.runtime.local_resource_url(self, statici18n_js_url))
+
+        frag.add_javascript(self.resource_string("static/js/src/filesmanager.js"))
+        frag.initialize_js('FilesManagerXBlock')
+        return frag
+
+    @XBlock.json_handler
+    def get_directories(self, data, suffix=''):
+        """Return the list of directories."""
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
+    @XBlock.json_handler
+    def clear_directories(self, data, suffix=''):
+        """Clear the list of directories without removing files from course assets."""
+        self.directories = []
+        self.incremental_directory_id = 0
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
+    @XBlock.json_handler
+    def get_content(self, data, suffix=''):
+        """Return the content of a directory."""
+        path = data.get("path")
+        if not path:
+            return {
+                "status": "error",
+                "message": "Path not found",
+            }
+        content, _, _ = self.get_content_by_path(path)
+        return {
+            "status": "success",
+            "content": content,
+        }
+
+    @XBlock.json_handler
+    def add_directory(self, data, suffix=''):
+        """Add a directory to a target directory."""
+        directory_name = data.get("name")
+        path = data.get("path")
+        target_directory = self.get_target_directory(path)
+        if not target_directory:
+            return {
+                "status": "error",
+                "message": "Target directory not found",
+            }
+        target_directory.append(
+            {
+                "name": directory_name,
+                "type": "directory",
+                "path": f"{path}/{directory_name}" if path else directory_name,
+                "metadata": {
+                    "id": self.incremental_directory_id,
+                },
+                "children": [],
+            }
+        )
+        self.incremental_directory_id += 1
```

Comment:
If we're using a unique path to identify content, I don't think an ID for directories would be needed.

Reply:
I'll be removing this. Thanks!

Reply:
Done, you can check again